cmd/kubelet/app/options/options.go

```diff
@@ -76,6 +76,12 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.TLSPrivateKeyFile, "tls-private-key-file", s.TLSPrivateKeyFile, "File containing x509 private key matching --tls-cert-file.")
 	fs.StringVar(&s.CertDirectory, "cert-dir", s.CertDirectory, "The directory where the TLS certs are located (by default /var/run/kubernetes). "+
 		"If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored.")
+	fs.StringVar(&s.BootstrapAuthToken, "bootstrap-auth-token", s.BootstrapAuthToken, ""+
```

Comment:
The kubeconfig already has exactly the information you need - the API endpoint, any associated CA info to connect securely to the API host, and the client credentials. Additionally, it allows for the credentials to be a bearer token, user/password, or client cert.  I don't see the benefit of having two sets of credentials and API location info, one used for bootstrapping only and limited to a bearer token.